leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java

```diff
@@ -513,6 +534,19 @@ public static void main(final String[] args) {
             }
         }
 
+        CertificateUsage certificateUsage = CertificateUsage.DOMAIN_ISSUER_CERTIFICATE;
+        if (cl.hasOption("cu")) {
+            certificateUsage = CertificateUsage.fromCode(Integer.parseInt(cl.getOptionValue("cu")));
+            if (trustStore.isEmpty()) {
+                if (certificateUsage == CertificateUsage.SERVICE_CERTIFICATE_CONSTRAINT) {
+                    System.err.println(
+                            "You need to set a truststore when you are using \"service_certificate_constraint\" usage");
+                    formatter.printHelp(USAGE, options);
+                    return;
```

Comment:
If I'm not wrong only SERVICE_CERTIFICATE_CONSTRAINT need a truststore as other mode can use the server certificate as truststore. 

(For CA constraint I think it should no be used as truststore too : see https://github.com/eclipse/leshan/issues/936)

Reply:
Need to have a bit of fresher mind in order to think this.

Reply:
@dachaac, the PR is merged but please do not hesitate to give me feedback about this here or at #936.